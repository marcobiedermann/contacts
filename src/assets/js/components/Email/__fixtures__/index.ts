const props = {
  value: 'john.doe@gmail.com',
};

export default props;
