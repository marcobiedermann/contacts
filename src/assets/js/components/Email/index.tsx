import React, { FC } from 'react';

export interface EmailProps {
  value: string;
}

const Email: FC<EmailProps> = (props) => {
  const { value } = props;

  return <a href={`mailto:${value}`}>{value}</a>;
};

export default Email;
