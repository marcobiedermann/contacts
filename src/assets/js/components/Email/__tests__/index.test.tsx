import { render } from '@testing-library/react';
import React from 'react';
import Email from '..';
import props from '../__fixtures__';

describe('email component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Email {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
