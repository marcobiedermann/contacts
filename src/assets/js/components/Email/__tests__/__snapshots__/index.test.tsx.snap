// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email component should render correctly 1`] = `
<DocumentFragment>
  <a
    href="mailto:john.doe@gmail.com"
  >
    john.doe@gmail.com
  </a>
</DocumentFragment>
`;
